ext/gettext/gettext.c

```diff
@@ -190,6 +190,11 @@ PHP_FUNCTION(bindtextdomain)
 
 	PHP_GETTEXT_DOMAIN_LENGTH_CHECK(1, ZSTR_LEN(domain))
 
+	if (ZSTR_VAL(domain)[0] == '\0') {
+		zend_argument_value_error(1, "cannot be empty");
+		RETURN_THROWS();
+	}
+
```

Comment:
It seems to me that `domain` as well as `dir` need to be treated as paths, i.e. neither must contain any NUL byte.

Reply:
yes good suggestion, I ll give it a try